packages/gestalt/src/UpsellForm.js

```diff
@@ -0,0 +1,47 @@
+// @flow strict
+import React, { type Node } from 'react';
+import Button from './Button.js';
+import Flex from './Flex.js';
+import { type AbstractEventHandler } from './AbstractEventHandler.js';
+import useResponsiveMinWidth from './useResponsiveMinWidth.js';
+
+type Props = {|
+  children: Node,
+  onSubmit: AbstractEventHandler<
+    | SyntheticMouseEvent<HTMLButtonElement>
+    | SyntheticMouseEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLAnchorElement>
+    | SyntheticKeyboardEvent<HTMLButtonElement>,
+  >,
+  submitButtonText: string,
+  submitButtonAccessibilityLabel: string,
+  submitButtonDisabled?: boolean,
+|};
+
+export default function UpsellForm({
+  children,
+  onSubmit,
+  submitButtonText,
+  submitButtonAccessibilityLabel,
+  submitButtonDisabled,
+}: Props): Node {
+  const responsiveMinWidth = useResponsiveMinWidth();
+
+  return (
+    <form onSubmit={onSubmit} style={{ width: '100%' }}>
+      <Flex gap={2} direction={responsiveMinWidth === 'xs' ? 'column' : 'row'} wrap>
+        <Flex.Item flex="grow">{children}</Flex.Item>
+        <Flex.Item>
+          <Button
+            accessibilityLabel={submitButtonAccessibilityLabel}
+            color="red"
+            disabled={submitButtonDisabled}
+            inline={responsiveMinWidth !== 'xs'}
```

Comment:
I haven't played with the code, but what if we inline by default? It might prevent the 100%-width button in xs breakpoint.

Reply:
<img width="461" alt="Screen Shot 2021-02-25 at 9 56 06 AM" src="https://user-images.githubusercontent.com/46180742/109195708-b5bb3a00-774f-11eb-8cb5-a6f4ac56e2d4.png">

Definitely needed!

Reply:
https://codesandbox.io/s/example-applying-flex-properties-to-children-forked-rpik6?file=/example.js

Reply:
ignore if we go with full width buttons.

Reply:
We actually do want to expand the button, so resolving this conversation!